homeassistant/components/netatmo/camera.py

```diff
@@ -46,59 +46,47 @@ async def async_setup_entry(hass, entry, async_add_entities):
         _LOGGER.info(
             "Cameras are currently not supported with this authentication method"
         )
-        return
 
     data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
 
     await data_handler.register_data_class(
         CAMERA_DATA_CLASS_NAME, CAMERA_DATA_CLASS_NAME, None
     )
+    data_class = data_handler.data.get(CAMERA_DATA_CLASS_NAME)
 
-    if CAMERA_DATA_CLASS_NAME not in data_handler.data:
+    if not data_class or data_class.raw_data == {}:
         raise PlatformNotReady
 
     async def get_entities():
         """Retrieve Netatmo entities."""
-
-        if not data_handler.data.get(CAMERA_DATA_CLASS_NAME):
-            return []
-
-        data_class = data_handler.data[CAMERA_DATA_CLASS_NAME]
-
         entities = []
-        try:
-            all_cameras = []
-            for home in data_class.cameras.values():
-                for camera in home.values():
-                    all_cameras.append(camera)
-
-            for camera in all_cameras:
-                _LOGGER.debug("Adding camera %s %s", camera["id"], camera["name"])
-                entities.append(
-                    NetatmoCamera(
-                        data_handler,
-                        camera["id"],
-                        camera["type"],
-                        camera["home_id"],
-                        DEFAULT_QUALITY,
-                    )
+        all_cameras = []
+
+        for home in data_class.cameras.values():
+            for camera in home.values():
+                all_cameras.append(camera)
+
+        for camera in all_cameras:
+            _LOGGER.debug("Adding camera %s %s", camera["id"], camera["name"])
```

Comment:
We can log all entities once after collecting them in the list instead. The entity has a repr magic method.